plc4j/api/src/main/java/org/apache/plc4x/java/api/value/PlcValues.java

```diff
@@ -355,6 +356,28 @@ public static PlcValue of(Map<String, PlcValue> map) {
         return new PlcStruct(map);
     }
 
+    private static PlcValue constructorHelper(Constructor<?> constructor, Object value) {
+        try {
+            return (PlcValue) constructor.newInstance(value);
+        } catch (InstantiationException | IllegalAccessException | InvocationTargetException e) {
+            throw new PlcIncompatibleDatatypeException(value.getClass());
+        }
+    }
+
+    public static PlcValue of(Object[] values, Class clazz) {
+        //Encode values to the type defined in clazz
+        try {
+            Constructor<?> constructor = clazz.getDeclaredConstructor(values[0].getClass());
+            if(values.length == 1) {
+                return ((PlcValue) constructor.newInstance(values[0]));
+            } else {
+                return PlcValues.of(Arrays.stream(values).map(value -> (constructorHelper(constructor, value))).collect(Collectors.toList()));
+            }
```

Comment:
One thing I've learnt recently, is that exceptions are extremely expensive, as creating the stack trace is expensive. is this code that is likely to produce any of "InstantiationException | IllegalAccessException | InvocationTargetException | NoSuchMethodException" in normal execution? If yes, we should probably change this and other parts where we use similar patterns to check first instead of silently catching exceptions.

Reply:
The exception would be raised if there wasn't a class (PlcBYTE, PlcINT, etc..) that matched the datatype that was passed. It shouldn't be raised in normal operation.